core/controllers/reader.py

```diff
@@ -294,6 +298,96 @@ def get(self, exploration_id):
+
+    REQUIRE_PAYLOAD_CSRF_CHECK = False
+
+    def _does_exp_issue_exist_in_exp_issues_model(
+            self, state_name_present, issue_customization_args,
+            customization_args):
+        """Checks whether the ExplorationIssue of the playthrough to be stored
+        already exists in the unresolved issues of the ExplorationIssuesModel.
+
+        Args:
+            state_name_present: bool. 'state_name' exists in
+                customization args if True, otherwise 'state_names' does.
+            issue_customization_args: dict. The customization dict for one
+                particular issue from the unresolved issues list.
+            customization_args: dict. The customization dict for the issue of
+                the playthrough to be stored.
+        """
+        if state_name_present:
+            if issue_customization_args['state_name'] == customization_args[
+                    'state_name']:
+                return True
+        else:
+            if issue_customization_args['state_names'] == customization_args[
+                    'state_names']:
+                return True
+        return False
+
+    @acl_decorators.can_play_exploration
+    def post(self, exploration_id):
+        """Handles POST requests.
+
+        Args:
+            exploration_id: str. The ID of the exploration.
+        """
+        exp_issue_dict = self.payload.get('exp_issue')
+        playthrough_data = self.payload.get('playthrough_data')
+
+        exp_issues_model = stats_models.ExplorationIssuesModel.get(
+            exploration_id)
+        exp_issues = stats_services.get_exp_issues_from_model(exp_issues_model)
+
+        playthrough_id = stats_models.PlaythroughModel.create(
+            playthrough_data['exp_id'],
+            playthrough_data['exp_version'],
+            playthrough_data['issue_id'],
+            playthrough_data['issue_customization_args'],
+            playthrough_data['playthrough_actions'],
+            playthrough_data['is_valid'])
+
+        customization_args = exp_issue_dict['customization_args']
+        # Boolean to identify difference between customization args that have
+        # either 'state_name' or 'state_names'.
+        state_name_in_customization_args = 'state_name' in customization_args
+
+        issue_found = False
+        for index, issue in enumerate(exp_issues.unresolved_issues):
+            if issue['issue_id'] == exp_issue_dict['issue_id']:
+                issue_customization_args = issue['issue_customization_args']
+                # This check checks if the state name or the list of state names
+                # (depending on the issue type) are identical.
+                if self._does_exp_issue_exist_in_exp_issues_model(
+                        state_name_in_customization_args,
+                        issue_customization_args, customization_args):
+                    if len(issue['playthrough_ids']) < (
```

Comment:
nit: move the parenthesis to wrap the entire expression:

```python
if (len(issue['playthrough_ids']) <
        feconf.MAX_PLAYTHROUGHS_FOR_ISSUE):
```

Reply:
Done.